### PR TITLE
[WIP] Front end design & cleanup

### LINKED ISSUE #11: need better frontend design
maybe @RichardLitt can help here?

### PR DESCRIPTION
**Pull request by Dignifiedquire**
Hopefully closes #11 soon.

Current version:

screen shot 2015-12-09 at 21 28 15

### Todo

- [ ] Design
  - [x] Base styles
  - [ ] Implement scroll spy for navigation
  - [ ] Dividers
  - [ ] Logo 
- [ ] Development
  - [x] Use webpack + harpjs for building
  - [x] Live reloading all the things using browsersync
  - [x] Linting with standard
  - [ ] Pull data from json/markdown files
- [ ] Build step
  - [ ] Integrate `harp compile` into `make`
  - [ ] Integrate `webpack` build into `make`
- [ ] Developer docs

---

You can run this via

```bash
$ npm install
$ npm i -g gulp
$ gulp serve
$ open http://localhost:3000
```

### CODE REVIEW

File: site/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+## Development
+
+Harpjs setup with
+
+* Live reload using browser-sync
+* JavaScript
+  * bundeling using webpack
+  * transpiling using babel
+* Linting using eslint
```
Comment (jbenet) [inline site/README.md]: are these required? why can't we keep these out? it makes it daunting to modify _websites_ when suddenly you have to know a ton more tools and know how to debug them when something goes wrong.

there's a lot to be said for simplicity and reducing cognitive complexity to the user.

**Comment by jbenet** [comment]
@Dignifiedquire this looks fantastic :+1: :clap: i think this is the way to go.

**Comment by jbenet** [comment]
@Dignifiedquire i'm super happy with how this turned out. thank you for taking all the feedback and finding a great way to satisfy all the crazy constraints, and still make it look great :) :+1:

File: site/README.md (hunk 1 above)
Comment (Dignifiedquire) [inline site/README.md]: There are a couple of major points why I think we need them

* Manage dependencies using npm
* Consistency with other browser based modules like `js-ipf-api`
* Make my life easier when working on the code (livereload is really a must for web dev for me)

The only thing that might not be needed is harpjs, though it was just a lot faster for me to use something working like that, than implementing all the tooling manually.

For me this is a lot simpler and more straight forward setup, especially while working with it than it was before. Also this enables me to add things like proper minification and asset bundeling which we need to have to ship this.

**Comment by Dignifiedquire** [comment]
> @Dignifiedquire this looks fantastic   i think this is the way to go.

Thanks, happy to hear that :tada: :balloon:

File: site/README.md (hunk 1 above)
Comment (jbenet) [inline site/README.md]: Ok sounds good. let's ship it

**Comment by jbenet** [comment]
@Dignifiedquire ahh sorry i merged too fast-- just recalled the open checkboxes. It's ok, maybe just continue with a PR on top.

**Comment by Dignifiedquire** [comment]
@jbenet no problem, will continue to create prs for the rest of the issues

**Pull request merged**